lib/filterx/expr-compound.c

```diff
@@ -84,6 +84,11 @@ _eval_exprs(FilterXCompoundExpr *self, FilterXObject **result)
   for (gint i = 0; i < self->exprs->len; i++)
     {
       filterx_object_unref(*result);
+      FilterXEvalContext *context = filterx_eval_get_context();
+
+      if (context->eval_control_type != FXC_NOTSET)
```

Comment:
I think that "not notset" is not what triggers the short circuit, but "drop" does, so I would check for that explicitly. We  might forget to check this code if a new control modifier gets introduced.